compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDialect.cpp

```diff
@@ -22,9 +26,120 @@ struct AMDAIEDialectOpAsmInterface : public OpAsmDialectInterface {
   }
 };
 
+/// Parse an instance of a type registered to the AMDAIE dialect.
+/// Parse an AMDAIE type in the following forms:
+///   AMDAIE-type
+///         ::= `logicalobjectfifo` `<` type `>`
+static OptionalParseResult amdaieTypeParser(DialectAsmParser &parser,
```

Comment:
Is there a reason why you used a custom type parser and printer instead of just doing `assemblyFormat` (https://mlir.llvm.org/docs/DefiningDialects/AttributesAndTypes/#adding-a-new-attribute-or-type-definition) . I think its great you went and looked at how to do it. It is useful to have this know how, but its easier to maintain with  just writing the assembly format.

Reply:
No, initially, I had an unsuccessful try with tablegen, probably due to unfamiliarity, and then  I based the implementation on the mlir-aie objectfifo type that uses this more verbose style. I created a new PR to use tablegen instead: https://github.com/nod-ai/iree-amd-aie/pull/270